src/components/BulletCursorOverlay.tsx

```diff
@@ -1,375 +1,157 @@
+    if (!state.cursor) {
+      return { simplePath: null, thoughtId: null, childrenThoughts: null, isInContextView: false }
+    }
 
-  return (
-    <div
-      aria-label='placeholder-ctx-breadcrumbs-outer-container'
-      className={css({
-        marginLeft: 'calc(1.3em - 14.5px)',
-        minHeight: '1em',
-        visibility: 'hidden',
-        marginBottom: '-0.25em', // Tighten up the space between the context-breadcrumbs and the thought (similar to the space above a note).
-        paddingTop: '0.5em', // Use padding-top instead of margin-top to ensure this gets included in the dynamic height of each thought. Otherwise the accumulated y value will not be correct.
-      })}
-      style={{
-        fontSize: '0.867em',
-        marginTop: '0.533em',
-      }}
-    >
-      {isRoot(simplePath) ? null : (
-        <span
-          style={{
-            wordBreak: 'break-word',
-            textDecoration: 'none',
-            WebkitTextStrokeWidth: '0.05em',
+    const path = simplifyPath(state, state.cursor)
+    const id = head(state.cursor)
+    const children = getAllChildren(state, id)
+      .map(childId => getThoughtById(state, childId))
+      .filter(Boolean)
+    const contextView = isContextViewActive(state, parentOf(state.cursor))
 
-            height: '1em',
-            margin: '-0.5em',
-            padding: '0.5em',
-            ...(isHaveMultipleAncestors
-              ? {
-                  lineHeight: '16px',
-                }
-              : {}),
-          }}
-          dangerouslySetInnerHTML={{ __html: '&ZeroWidthSpace;' }}
-        ></span>
-      )}
-    </div>
-  )
-}
-
-/**
- * PlaceholderTreeNode is a component used to mimic behavior of TreeNode.
- * Any position changes from one Thought to another will be animated within this component.
- */
-function PlaceholderTreeNode({
-  children,
-  isTableCol1 = false,
-  x,
-  y,
-  width,
-}: {
-  children?: React.ReactNode
-  isTableCol1?: boolean
-  x: number
-  y: number
-  width: number
-}) {
-  const outerDivStyle = {
-    left: x,
-    top: y,
-    // Table col1 uses its exact width since cannot extend to the right edge of the screen.
-    // All other thoughts extend to the right edge of the screen. We cannot use width auto as it causes the text to wrap continuously during the counter-indentation animation, which is jarring. Instead, use a fixed width of the available space so that it changes in a stepped fashion as depth changes and the word wrap will not be animated. Use x instead of depth in order to accommodate ancestor tables.
-    // 1em + 10px is an eyeball measurement at font sizes 14 and 18
-    // (Maybe the 10px is from .content padding-left?)
-    width: isTableCol1 ? width : `calc(100% - ${x}px + 1em + 10px)`,
-    textAlign: isTableCol1 ? ('right' as const) : undefined,
-  }
+    return { simplePath: path, thoughtId: id, childrenThoughts: children, isInContextView: contextView }
```

Comment:
Since you are returning an object literal, this will return a new object reference every time, regardless of whether any of the values have changed, i.e. `({}) !== ({})`. This has the unfortunate effect of causing the component to re-render on every `State` change.

To do a deep equal comparison and only re-render when a value changes, pass lodash's `isEqual` as the equality function:

```
useSelector(fn, isEqual)
```